src/dagmc/DagMC.hpp

```diff
@@ -0,0 +1,460 @@
+#ifndef MOABMC_HPP
+#define MOABMC_HPP
+
+#include "MBTagConventions.hpp"
+#include "moab/CartVect.hpp"
+#include "moab/Range.hpp"
+#include "moab/Core.hpp"
+#include "moab/GeomUtil.hpp"
+#include "moab/FileOptions.hpp"
+#include "moab/Interface.hpp"
+#include "moab/GeomTopoTool.hpp"
+#include "moab/GeomQueryTool.hpp"
+#include "DagMCVersion.hpp"
+
+#include <vector>
+#include <map>
+#include <string>
+#include <assert.h>
+
+class RefEntity;
+
+struct DagmcVolData {
+  int mat_id;
+  double density, importance;
+  std::string comp_name;
+};
+
+
+namespace moab {
+
+class CartVect;
+
+/**\brief
+ *
+ * In sectction 1, the public interface you will find all the functions needed
+ * for problem setup. For the typical MC code, the order of function calls
+ * required to fully populate DAGMC ready to run are
+ *
+ *    1) DAG->load_file();
+ *    2) DAG->init_OBBTree();
+ *
+ * Modifications were made to init_OBBTree which allows the functions of
+ * init_OBBTree to be called without having used init_OBBTree. For example
+ * if you would like access to be able to call DAG->point_in_volume() but without
+ * having an implicit compliment you need only call
+ *
+ *   1) DAG->load_file();
+ *   2) DAG->setup_obb();
+ *
+ * Similarly, if you need access to problem indices only then, one may call
+ * load_file followed by setup_indices.
+ *
+ *   1) DAG->load_file();
+ *   2) DAG->setup_indices();
+ */
+
+class DagMC {
+ public:
+  // Constructor
+  DagMC(Interface* mb_impl = NULL, double overlap_tolerance = 0., double numerical_precision = .001);
+  // Destructor
+  ~DagMC();
+
+  /** Return the version of this library */
+  static float version(std::string* version_string = NULL);
+  /** Get subversion revision of this file (DagMC.hpp) */
+  static unsigned int interface_revision();
+
+  /* SECTION I: Geometry Initialization */
+
+  /**\brief Load a geometry description regardless of format
+   *
+   * This method will load the geometry file with name cfile.
+   * In case this is a solid model geometry file, it will pass
+   * the facet_tolerance option as guidance for the faceting engine.
+   *\param cfile the file name to be loaded
+   *\param facet_tolerance the faceting tolerance guidance for the faceting engine
+   *\return - MB_SUCCESS if file loads correctly
+   *        - other MB ErrorCodes returned from MOAB
+   *
+   * Note: When loading a prexisting file with an OBB_TREE tag, a number of unspoken
+   * things happen that one should be aware of.
+   *
+   * 1) The file is loaded and when we query the meshset, we find entities with the OBB_TREE tag
+   * 2) The OBBTreeTool assumes that any children of the entity being queried in a ray intersect sets
+   *     operation are fair game, the surface meshesets have triangles as members, but OBB's as children
```

Comment:
Grammar:  `but OBB's as children but no querying is done` -> `that OBBs have children but no querying is done`?